### PR TITLE
[Changed] New challenge

### PR DESCRIPTION
**Pull request by jose-aparici**
This pull requests remove everything but readme file and some images due to implement the new challenge approach

TODO:

- [x] Add expected time candidates should spend in the challenge
- [x] Invite to submit challenge via **private** repo sharing permissions.
- [x] Clean git / node requirements comments

### CODE REVIEW

**Comment by nico-creditas** [comment]
In summary, all this repo will contain is the Readme file with the instructions, mockups and endpoints that the candidates have to follow.

**Comment by nico-creditas** [comment]
This is how the challenge would be -> https://github.com/Creditas/frontend-challenge-vlc/tree/newReadme

File: README.md [hunk 1]
```diff
@@ -1,157 +1,160 @@
...
+![Mobile version](./mobile-layout.png)
 
 ## Development
 
 ### Prerequisites
 
 You will need to install on your machine:
-- [NodeJs](https://nodejs.org/en/) in order to run the application.
+
 - [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) in order to clone the repository containing this challenge and submit the pull request with your solution.
```
Comment (nico-creditas) [inline README.md:121]: If we're not expecting PRs then we can remove this section of cloning the repo.

**Review by joseahernandez — APPROVED** [review]
Congrats guys!! That challenge is much more understandable that the previous one and simplify the work to do 🚀

**Comment by aleixmorgadas** [comment]
Hi there 👋 

The challenge looks quite complete and straightforward to read 😄

I would give my 5 cents from the candidate point of view, seems a lot but indeed it isn't that much 😅 

I would add the amount of time we think they should dedicate to the challenge. Do we expect them to dedicate 4h or 8h? This will guide them to not over engineer the solution.

I would say that, meanwhile some developers might prefer to choose their own stack and setup it, others might like to have the setup already there so they can focus on the challenge itself instead of setting up all the project. Being conscious in the candidates time, I would prefer all time dedicated in the challenge itself IMHO.

In the mean time, I would suggest some of the Stacks that we would prefer to evaluate to guide them. Something like:

> Stack:
> - You are free to choose the stack and language that you feel comfortable with!
> - In case of doubt which one to choose, we prefer React over Vue or Angular since that's the stack we are currently using at Creditas

On the other hand, which is the reason of?:

> Please send us via email your solution into a zip file.

If I had to choose, I prefer them to invite me to a private repository so I can see how they solved the solution + the commits, you can see if they did TDD or Test Last for example based on the commits usually. I hope to not lose that information as a zip file.

Because I've not all the context about this new approach, I'm just sharing this based on previous experiences, they might or might not apply in this situation 🤷‍♂️ . I hope it helped in some way 😄

File: README.md [hunk 2]
```diff
@@ -1,157 +1,160 @@
...
 
-At Creditas we always work giving constructive feedback, therefore we will always dedicate the outmost attention to every solution submitted; contacting you back with the positive aspects of your submission and what could be improved. Just for that it is worth trying! :)
+At Creditas we always work giving constructive feedback, therefore we will always dedicate the outmost attention to every solution submitted, contacting you back with the positive aspects of your submission and what could be improved.
+
+Just for that it is worth trying! :)
 
 ## The challenge
 
-You have to improve the loan simulator implemented in this project.
+Cine Creditas is getting bigger and bigger and needs to create a self-service booking system.
```
Comment (rsicarelli) [inline README.md:15]: Cine Creditas FTW!

File: README.md [hunk 3]
```diff
@@ -1,157 +1,160 @@
...
 
-# Frontend Challenge - Creditas
+# Frontend Challenge
 
 This is a challenge designed to measure the candidate's expertise.
-We would like to clarify that we don't expect everyone to be able to finish the whole challenge, since it has been designed to cover several degrees of expertise, from junior to senior.
-Nevertheless, we expect anyone interested in becoming a tripulante in Creditas to give it a try.
 
-In this challenge,cd we focus both on code design and design patterns in JavaScript. The goal is to assess your experience in writing **code that is easy to maintain, loosely coupled and highly cohesive**.
+In this challenge,we focus both on code design and design patterns in JavaScript.
```
Comment (rsicarelli) [inline README.md:7]:
```suggestion
In this challenge, we focus both on code design and design patterns in JavaScript.
```

File: README.md [hunk 4]
```diff
@@ -1,157 +1,160 @@
...
 
-At Creditas we always work giving constructive feedback, therefore we will always dedicate the outmost attention to every solution submitted; contacting you back with the positive aspects of your submission and what could be improved. Just for that it is worth trying! :)
+At Creditas we always work giving constructive feedback, therefore we will always dedicate the outmost attention to every solution submitted, contacting you back with the positive aspects of your submission and what could be improved.
```
Comment (rsicarelli) [inline README.md:9]:
```suggestion
At Creditas, we value giving (and receiving) constructive feedback, therefore we will always dedicate the outmost attention to every solution submitted, contacting you back with the positive aspects of your submission and what could be improved.
```

File: README.md [hunk 5]
```diff
@@ -1,157 +1,160 @@
...
+
+Just for that it is worth trying! :)
```
Comment (rsicarelli) [inline README.md:11]:
```suggestion
Just for that, it is worth trying! :). We also ask you to help us to improve our hiring process, at the end of this file you will find a form allowing you to share your impressions with us
```

File: README.md [hunk 6]
```diff
@@ -1,157 +1,160 @@
...
 
 ## The challenge
 
-You have to improve the loan simulator implemented in this project.
+Cine Creditas is getting bigger and bigger and needs to create a self-service booking system.
 
-Both the interface as well as the styles are provided. The challenges include refactoring the code and implementing new functionalities (listed below).
+In order to make a booking, users must select a **film** and a **snack**.
```
Comment (rsicarelli) [inline README.md:17]:
```suggestion
In order to make a booking, our clients must select a **film** and a **snack**.
```

File: README.md [hunk 7]
```diff
@@ -1,157 +1,160 @@
...
 
-Feel free to componetize whatever you think should be componetized. Our only request is that you use only pure JavaScript, our beloved Vanilla. Do you accept the challenge?
+It is **important** to remark that our business model requires the user to select both elements ( a film and a snack )
```
Comment (rsicarelli) [inline README.md:19]:
```suggestion
It is **important** to remark that our business model requires the client to select both elements ( a film and a snack )
```

File: README.md [hunk 8]
```diff
@@ -1,157 +1,160 @@
...
 
-This is the simulator current state:
+Once the user has selected both items, they must confirm their selection and as a result a ticket will be displayed describing user's choice.
```
Comment (rsicarelli) [inline README.md:21]:
```suggestion
Once the client has selected both items, they must confirm their selection and as a result, a ticket will be displayed describing the client's choice.
```

File: README.md [hunk 9]
```diff
@@ -1,157 +1,160 @@
...
 
-<img style="display: block; margin: 0 auto;" src="./layout_en.png">
+Your mission will be to build a web application where the user can see both a list of films and a list of snacks, have the possibility to confirm their selection and display the corresponding generated ticket.
```
Comment (rsicarelli) [inline README.md:23]:
```suggestion
Your mission will be to build a web application where the client can see both a list of films and a list of snacks, have the possibility to confirm their selection and display the corresponding generated ticket.
```

File: README.md [hunk 10]
```diff
@@ -1,157 +1,160 @@
...
 
+Right now due to the traffic we are generating, we are averaging over 10 seconds to generate the ticket.
```
Comment (rsicarelli) [inline README.md:25]:
```suggestion
Right now due to the right amount of clients accessing our website, we are averaging over 10 seconds to generate the ticket.
```

File: README.md [hunk 11]
```diff
@@ -1,157 +1,160 @@
...
 
-### Expected functionalities
+As part of the challenge we need your help in order to reduce this delay as much as possible.
```
Comment (rsicarelli) [inline README.md:27]: Could you please provide further explanation about the reasoning of these criteria? 

I'm struggling to understand what exactly the candidate has to do in order to improve this 10 seconds delay. Since there is no additional code/implementation in the assignment, it might be super easy for the candidate to pass into these criteria, so I'm not sure if its worth to add this here

is there some kind of delay in the `POST: /api/films/confirm` and `POST: /api/snacks/confirm`?

File: README.md [hunk 12]
```diff
@@ -1,157 +1,160 @@
...
-***"Auto"*** or ***"Home"*** (the default option is ***"Auto"*** ).
+## Resources
 
-It should also follow the calculation rules bellow:
+### Endpoints
 
-- Financial Transaction Tax (FTT): 6.38%
-- Interest Rate: 2.34%
-- Maximum loan amount (maximum amount the user can borrow) 80% of the collateral's value. This means that when you change the collateral value, **the maximum for the loan amount slider should be recalculated too**
+- GET https://frontend-challenge-vlc-api.public.dev.creditas.io/api/films
```
Comment (rsicarelli) [inline README.md:33]: This endpoint returns only 3 items. Maybe worth adding a bit more items (maybe +10? +20?) to also check the performance/experience on the mobile version. If there are bad performance practices in place, the experience will be bad in mobile, having only 3 items it's hard to spot that

File: README.md [hunk 13]
```diff
@@ -1,157 +1,160 @@
...
-To display the help text, we provide two endpoints on localhost:4000 ( for more details you can check server.js file ):
-- 'api/question' which responses is 
-```
-{ text: 'Question: Can you help me?' }
-```
-- 'api/answer' which response is 
-```
-{ text: 'Answer: Yes sure.' }
-```
+The template requires some css rules in order to be displayed as proposed in the screenshots provided. Feel free to use any property you feel like to achieve so.
```
Comment (rsicarelli) [inline README.md:106]: Could you also add which items are mandatory to display? Maybe things like `image`, `price` and `title`

Even if it's super obvious for us, worth making it clear to the candidate as well

File: README.md (hunk 1 above)
Comment (rsicarelli) [reply README.md:121]: +1

File: README.md [hunk 14]
```diff
@@ -1,157 +1,160 @@
...
 ## Development
 
 ### Prerequisites
 
 You will need to install on your machine:
-- [NodeJs](https://nodejs.org/en/) in order to run the application.
+
 - [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) in order to clone the repository containing this challenge and submit the pull request with your solution.
 
 ### Configuration to run the environment
```
Comment (rsicarelli) [inline README.md:123]: Combined with @nico-creditas comment above, I think that we could remove this section as well

**Review by rsicarelli — COMMENTED** [review]
Good stuff people! Well done

I added some suggestions and questions. Feel free to use them or not 🚀

File: README.md (hunk 12 above)
Comment (rsicarelli) [reply README.md:33]: Also, what if we add random 2-3 broken images in the items in order to evaluate the ability to handle error cases?

File: README.md (hunk 11 above)
Comment (nico-creditas) [reply README.md:27]: @rsicarelli The thing is that the ticket needs information from the response of both endpoints. They can do these requests sequentially or in parallel. This is where the time will considerably change, since the requests take 5 seconds to respond.

Comment (nico-creditas) [reply README.md:27]: To clarify:

> is there some kind of delay in the POST: /api/films/confirm and POST: /api/snacks/confirm?

Yes, there is!

**Review by aleixmorgadas — APPROVED** [review]
Looks great to me. When do you plan to start sharing this challenge for new people being interviewed?

**Comment by nico-creditas** [comment]
Hi @aleixmorgadas ! Thanks for the review! We still have some updates to do in response to the feedback gathered here (we haven't forgotten about it 😄 ), and once we have time to commit those changes we'll be ready to share it!

File: README.md (hunk 11 above)
Comment (rsicarelli) [reply README.md:27]: Awesome, clear! You can resolve this comment if you wish 👌

File: README.md [hunk 15]
```diff
@@ -1,157 +1,144 @@
...
-### Dependencies
+### Stack
 
-This project uses some libraries to assist us with tests and running the application. Those are: 
-- [Webpack](https://webpack.js.org/)
-- [Babel](https://babeljs.io/)
-- [Jest](https://jestjs.io/)
-- [Eslint)[https://eslint.org/]
+- You are free to choose the stack and language that you feel comfortable with!
+- In case of doubt which one to choose, we prefer React over Vue or Angular since that's the stack we are currently using at Creditas
```
Comment (joseahernandez) [inline README.md:128]: I am not sure about that sentence. Indirectly we are forcing candidates to do the challenge in React because "that is what we prefer". Moreover, in the first sentence we are talking about choose the stack that you want and in that second one we are indicated 3 frameworks 🤔

Comment (aleixmorgadas) [reply README.md:128]: It might seem like forcing, but I see it more like an orientation. Still, I prefer to give a set of preferred frameworks.

In that scenario, a jQuery solution is totally valid and we might not be able to evaluate what we want to evaluate compared to a solution based on those frameworks.

I would choose based on what puts the candidate in a better position for success.

Comment (joseahernandez) [reply README.md:128]: Clearly we have different point of view of what is best for a candidate and it's great to have different opinions. This would be a great topic to discuss with a beer, if it weren't for the damn covid 😞

## tldr
For me is ok to have that sentence, but definitely one of the two sentences should be deleted, we can't say choose the stack you want and in the following sentence say we prefer React. 

## Why I don't think that is the best for a candidate?
The first point is that frontend isn't only React, Angular or Vue. Exists a lot of frameworks ember, meteor, mithril... Moreover exists a lot of great engineers that work with vanilla JS or even jQuery and most of them don't know anything about React, Angular or Vue. Really we want to discard that people? 

For me a great engineers isn't that who knows a framework. In fact, for me that is one of the things that I found least valuable when looking for a candidate. In a candidate I look to how she thinks, why she applies that solution to resolve that concrete problem? How the code is organize? Is she applying good pattern to resolve the problem? That are better things to look in a candidate better that if she can user one or another FW. That is the reason why I don't like to put anything about a framework in a challenge.

But as I said before, that is just my opinion and I don't have any problem if we decided to leave that mention to the frameworks. We can always receive feedback from candidates and iterate to improve our challenge 💪

Comment (aleixmorgadas) [reply README.md:128]: I totally understand your point of view.

Based on the discussion, I changed my mind and I would go first with the most freedom for the candidates instead of suggesting what to use (I assumed an issue without any fact backing it), and based on feedback we iterate 😄

Comment (nico-creditas) [reply README.md:128]: We are going to keep it open, and not hint about any specific framework.
We are not going to provide a setup of the challenge since the we want to give freedom to candidates to choose their stack.

Thanks for all your suggestions!
